diff --git a/node_modules/@ledgerhq/hw-transport-webusb/lib-es/TransportWebUSB.js b/node_modules/@ledgerhq/hw-transport-webusb/lib-es/TransportWebUSB.js
index b5e41c8..567ca59 100644
--- a/node_modules/@ledgerhq/hw-transport-webusb/lib-es/TransportWebUSB.js
+++ b/node_modules/@ledgerhq/hw-transport-webusb/lib-es/TransportWebUSB.js
@@ -50,7 +50,7 @@ var __generator = (this && this.__generator) || function (thisArg, body) {
     }
 };
 import Transport from "@ledgerhq/hw-transport";
-import hidFraming from "@ledgerhq/devices/hid-framing";
+import hidFraming from "@ledgerhq/devices/lib-es/hid-framing";
 import { identifyUSBProductId } from "@ledgerhq/devices";
 import { log } from "@ledgerhq/logs";
 import { TransportOpenUserCancelled, TransportInterfaceNotAvailable, TransportWebUSBGestureRequired, DisconnectedDeviceDuringOperation, DisconnectedDevice, } from "@ledgerhq/errors";
diff --git a/node_modules/@ledgerhq/hw-transport-webusb/lib/TransportWebUSB.js b/node_modules/@ledgerhq/hw-transport-webusb/lib/TransportWebUSB.js
index 1fc8954..4a79682 100644
--- a/node_modules/@ledgerhq/hw-transport-webusb/lib/TransportWebUSB.js
+++ b/node_modules/@ledgerhq/hw-transport-webusb/lib/TransportWebUSB.js
@@ -55,7 +55,7 @@ var __importDefault = (this && this.__importDefault) || function (mod) {
 };
 exports.__esModule = true;
 var hw_transport_1 = __importDefault(require("@ledgerhq/hw-transport"));
-var hid_framing_1 = __importDefault(require("@ledgerhq/devices/hid-framing"));
+var hid_framing_1 = __importDefault(require("@ledgerhq/devices/lib/hid-framing"));
 var devices_1 = require("@ledgerhq/devices");
 var logs_1 = require("@ledgerhq/logs");
 var errors_1 = require("@ledgerhq/errors");
