diff --git a/node_modules/@ledgerhq/hw-transport-webhid/lib/TransportWebHID.js b/node_modules/@ledgerhq/hw-transport-webhid/lib/TransportWebHID.js
index 5dbe29b..50fa199 100644
--- a/node_modules/@ledgerhq/hw-transport-webhid/lib/TransportWebHID.js
+++ b/node_modules/@ledgerhq/hw-transport-webhid/lib/TransportWebHID.js
@@ -71,7 +71,7 @@ var __importDefault = (this && this.__importDefault) || function (mod) {
 };
 exports.__esModule = true;
 var hw_transport_1 = __importDefault(require("@ledgerhq/hw-transport"));
-var hid_framing_1 = __importDefault(require("@ledgerhq/devices/hid-framing"));
+var hid_framing_1 = __importDefault(require("@ledgerhq/devices/lib/hid-framing"));
 var devices_1 = require("@ledgerhq/devices");
 var logs_1 = require("@ledgerhq/logs");
 var errors_1 = require("@ledgerhq/errors");
