diff --git a/node_modules/axios/lib/core/AxiosError.js b/node_modules/axios/lib/core/AxiosError.js
index 52c806a..abd5cc5 100644
--- a/node_modules/axios/lib/core/AxiosError.js
+++ b/node_modules/axios/lib/core/AxiosError.js
@@ -14,12 +14,14 @@ var utils = require('../utils');
  */
 function AxiosError(message, code, config, request, response) {
   Error.call(this);
-  this.message = message;
-  this.name = 'AxiosError';
-  code && (this.code = code);
-  config && (this.config = config);
-  request && (this.request = request);
-  response && (this.response = response);
+  if (this) {
+    this.message = message;
+    this.name = 'AxiosError';
+    code && (this.code = code);
+    config && (this.config = config);
+    request && (this.request = request);
+    response && (this.response = response);
+  }
 }
 
 utils.inherits(AxiosError, Error, {
