diff --git a/node_modules/bitcoinjs-lib/src/address.js b/node_modules/bitcoinjs-lib/src/address.js
index 2712168..089f663 100644
--- a/node_modules/bitcoinjs-lib/src/address.js
+++ b/node_modules/bitcoinjs-lib/src/address.js
@@ -46,9 +46,10 @@ function fromBase58Check(address) {
   // TODO: 4.0.0, move to "toOutputScript"
   if (payload.length < 21) throw new TypeError(address + ' is too short');
   if (payload.length > 21) throw new TypeError(address + ' is too long');
-  const version = payload.readUint8(0);
+  const version = payload.slice(0,1)[0];
   const hash = payload.slice(1);
   return { version, hash };
+  
 }
 exports.fromBase58Check = fromBase58Check;
 function fromBech32(address) {
@@ -122,7 +123,9 @@ function toOutputScript(address, network) {
   let decodeBech32;
   try {
     decodeBase58 = fromBase58Check(address);
-  } catch (e) {}
+  } catch (e) {
+  console.log("🚀 ~ file: address.js:127 ~ toOutputScript ~ e:", e)
+  }
   if (decodeBase58) {
     if (decodeBase58.version === network.pubKeyHash)
       return payments.p2pkh({ hash: decodeBase58.hash }).output;
