diff --git a/node_modules/bitcoinjs-lib/src/address.js b/node_modules/bitcoinjs-lib/src/address.js
index e15c55e..2f014cb 100644
--- a/node_modules/bitcoinjs-lib/src/address.js
+++ b/node_modules/bitcoinjs-lib/src/address.js
@@ -12,7 +12,7 @@ function fromBase58Check(address) {
   // TODO: 4.0.0, move to "toOutputScript"
   if (payload.length < 21) throw new TypeError(address + ' is too short');
   if (payload.length > 21) throw new TypeError(address + ' is too long');
-  const version = payload.readUInt8(0);
+  const version = payload.slice(0,1)[0];
   const hash = payload.slice(1);
   return { version, hash };
 }
