diff --git a/node_modules/@ledgerhq/react-native-hw-transport-ble/lib/BleTransport.d.ts b/node_modules/@ledgerhq/react-native-hw-transport-ble/lib/BleTransport.d.ts
index 86f3f2d..d01352f 100644
--- a/node_modules/@ledgerhq/react-native-hw-transport-ble/lib/BleTransport.d.ts
+++ b/node_modules/@ledgerhq/react-native-hw-transport-ble/lib/BleTransport.d.ts
@@ -65,5 +65,6 @@ export default class BluetoothTransport extends Transport {
     write: (buffer: Buffer, txid?: string | null | undefined) => Promise<void>;
     close(): Promise<void>;
 }
-export {};
+import { BleManager } from "react-native-ble-plx";
+export declare function bleManagerInstance(): BleManager;
 //# sourceMappingURL=BleTransport.d.ts.map
\ No newline at end of file
diff --git a/node_modules/@ledgerhq/react-native-hw-transport-ble/lib/BleTransport.js b/node_modules/@ledgerhq/react-native-hw-transport-ble/lib/BleTransport.js
index 366f0d3..0f1644b 100644
--- a/node_modules/@ledgerhq/react-native-hw-transport-ble/lib/BleTransport.js
+++ b/node_modules/@ledgerhq/react-native-hw-transport-ble/lib/BleTransport.js
@@ -672,6 +672,7 @@ var BluetoothTransport = /** @class */ (function (_super) {
     return BluetoothTransport;
 }(hw_transport_1["default"]));
 exports["default"] = BluetoothTransport;
+exports.bleManagerInstance = bleManagerInstance;
 var bleErrorToHwTransportError = new Map([
     [react_native_ble_plx_1.BleErrorCode.ScanStartFailed, errors_1.HwTransportErrorType.BleScanStartFailed],
     [
